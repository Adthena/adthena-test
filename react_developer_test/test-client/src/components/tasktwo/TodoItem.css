.todo {
  text-decoration: line-through;
}
.todo-item {
  margin-top: 15px;
}
