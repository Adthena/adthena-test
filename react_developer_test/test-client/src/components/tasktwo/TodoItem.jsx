
import React from 'react';

const TodoItem = (props) => {
    return (
        <li className={!props.complete ? 'todo' : ''}>
            {props.title}
            {props.userId}
        </li>)
};

export default TodoItem;
