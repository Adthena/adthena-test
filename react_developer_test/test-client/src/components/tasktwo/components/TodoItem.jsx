
import React from 'react';
import PropTypes from 'prop-types'

const TodoItem = (props) => {
    return (
        <li className={!props.completed ? 'todo' : ''}>
            {props.title}
        </li>)
};

TodoItem.propTypes = {
    title: PropTypes.string.isRequired,
    completed: PropTypes.bool.isRequired
};

export default TodoItem;
