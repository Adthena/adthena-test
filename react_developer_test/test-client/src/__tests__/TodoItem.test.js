import React from 'react';
import { render } from "enzyme";
import TodoItem from '../components/tasktwo/components/TodoItem';

it('renders correctly', () => {
    const item = {
        "userId": 1,
        "id": 1,
        "title": "delectus aut autem",
        "completed": false
    };
    const rendered = render(<TodoItem {...item} />)
    expect(rendered).toMatchSnapshot();
})
