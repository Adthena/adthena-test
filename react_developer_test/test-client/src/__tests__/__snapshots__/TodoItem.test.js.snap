// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  class="todo"
>
  delectus aut autem
</li>
`;
